Remove Agricultural Bulletin functionality

bozaah opened the pull request:
**Breaking changes**:  
- Removed ('defunct') functions `get_ag_bulletin()` and `parse_ag_bulletin()` because BOM's agricultural bulletin service is no longer available since early February 2025.
- Updated tests to reflect the removal of these functions.
- Proposed version updated to ~~v1.1.0~~ v2.0.0

**Bug fixes**:  
- Fixed issues with `get_dpird_summaries()`, including updating tests to dynamically use the previous year instead of fixed dates.
- Resolved test bugs in `find_stations_in()` where the function did not return the correct number of stations due to recent changes in the weather station infrastructure.
- Updated vignettes to remove defunct functionality.

**Additional fixes and improvements**:  
- Re-documented package to reflect changes
- Updated test cassete fixtures
- Updated BOM-related vignette
- Updated `WORDLIST` for spelling checks
- Check are passing locally and via GitHub Actions.

bozaah commented:
All tests and check are passing locally.

bozaah commented:
@adamhsparks, is it reasonable to bump the current version to v1.1.0? Reading about versioning semantics, it is suggested that we should bump it to v2.0.0 due to the breaking changes. However, these changes are only minor functionality, so I have proposed v1.1.0. 

After that, I will create a new release tag and proceed with the release.

adamhsparks commented:
Removing functionality totally breaks backwards compatibility. Yes, it seems funny to jump to 2.0.0 so suddenly after the initial release, but in reality that's what you should do here. A minor version, like you've proposed, would be acceptable if you only added new functionality, not removed which could break workflows so a major release is called for.

bozaah commented:
Thanks @adamhsparks agreed! 

I have updated the description and news.md to reflect the breaking changes. I have rebuilt the vignettes, re-doc'ed the package and updated files related to the changes. Tests and checks are passing locally and with the GitHub Action R-CMD checks. After the merge, I will create a new release. 

After the changes move through the r-universe checks/tests, I will submit the new version to CRAN.
